I could not complete the build on fedora.

I'm on fedora 39 and I installed cargo using rustup.
I ran `cargo install wayland-pipewire-idle-inhibit`

I got two errors because I didn't have pipewire-devel and clang installed. After installing the dependencies I got this error:
```sh
   Compiling xdg v2.5.2
   Compiling zbus v4.1.2
   Compiling pipewire v0.7.2
error[E0432]: unresolved import `spa::data`
 --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/buffer.rs:3:10
  |
3 | use spa::data::Data;
  |          ^^^^ could not find `data` in `spa`

error[E0432]: unresolved import `spa::flags`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/channel.rs:74:10
   |
74 | use spa::flags::IoFlags;
   |          ^^^^^ could not find `flags` in `spa`

error[E0432]: unresolved imports `spa::dict`, `spa::result`, `spa::AsyncSeq`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/core_.rs:18:11
   |
18 | use spa::{dict::ForeignDict, result::SpaResult, spa_interface_call_method, AsyncSeq};
   |           ^^^^               ^^^^^^ could not find `result` in `spa`       ^^^^^^^^ no `AsyncSeq` in the root
   |           |
   |           could not find `dict` in `spa`
   |
   = help: consider importing one of these items instead:
           crate::spa::utils::result::AsyncSeq
           spa::utils::result::AsyncSeq

error[E0432]: unresolved import `spa::dict`
 --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/link.rs:9:10
  |
9 | use spa::dict::ForeignDict;
  |          ^^^^ could not find `dict` in `spa`

error[E0432]: unresolved imports `spa::flags`, `spa::result`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/loop_.rs:15:11
   |
15 | use spa::{flags::IoFlags, result::SpaResult, spa_interface_call_method};
   |           ^^^^^           ^^^^^^ could not find `result` in `spa`
   |           |
   |           could not find `flags` in `spa`

error[E0432]: unresolved import `spa::ReadableDict`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/main_loop.rs:10:5
   |
10 | use spa::ReadableDict;
   |     ^^^^^^^^^^^^^^^^^ no `ReadableDict` in the root

error[E0432]: unresolved import `spa::dict`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/node.rs:14:11
   |
14 | use spa::{dict::ForeignDict, pod::Pod, spa_interface_call_method};
   |           ^^^^ could not find `dict` in `spa`

error[E0432]: unresolved import `crate::spa::Direction`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/port.rs:11:5
   |
11 |     spa::Direction,
   |     ^^^^^^^^^^^^^^ no `Direction` in the root
   |
   = help: consider importing one of these items instead:
           crate::spa::utils::Direction
           spa::utils::Direction

error[E0432]: unresolved import `spa::dict`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/port.rs:14:11
   |
14 | use spa::{dict::ForeignDict, pod::Pod, spa_interface_call_method};
   |           ^^^^ could not find `dict` in `spa`

error[E0432]: unresolved import `spa::dict`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:18:11
   |
18 | use spa::{dict::ForeignDict, prelude::*};
   |           ^^^^ could not find `dict` in `spa`

error[E0432]: unresolved import `spa::result`
 --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/stream.rs:9:10
  |
9 | use spa::result::SpaResult;
  |          ^^^^^^ could not find `result` in `spa`

error[E0433]: failed to resolve: could not find `dict` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/core_.rs:150:52
    |
150 |     pub fn create_object<P: ProxyT, D: crate::spa::dict::ReadableDict>(
    |                                                    ^^^^ could not find `dict` in `spa`

error[E0433]: failed to resolve: could not find `hook` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/core_.rs:228:14
    |
228 |         spa::hook::remove(*self.listener);
    |              ^^^^ could not find `hook` in `spa`
    |
help: consider importing one of these items
    |
4   + use crate::spa::utils::hook;
    |
4   + use spa::utils::hook;
    |
help: if you import `hook`, refer to it directly
    |
228 -         spa::hook::remove(*self.listener);
228 +         hook::remove(*self.listener);
    |

error[E0412]: cannot find type `Error` in crate `spa`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/error.rs:14:27
   |
14 |     SpaError(#[from] spa::Error),
   |                           ^^^^^ not found in `spa`
   |
help: consider importing one of these items
   |
4  + use anyhow::Error;
   |
4  + use core::error::Error;
   |
4  + use core::fmt::Error;
   |
4  + use crate::Error;
   |
     and 6 other candidates
help: if you import `Error`, refer to it directly
   |
14 -     SpaError(#[from] spa::Error),
14 +     SpaError(#[from] Error),
   |

error[E0433]: failed to resolve: could not find `hook` in `spa`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/link.rs:66:14
   |
66 |         spa::hook::remove(*self.listener);
   |              ^^^^ could not find `hook` in `spa`
   |
help: consider importing one of these items
   |
1  + use crate::spa::utils::hook;
   |
1  + use spa::utils::hook;
   |
help: if you import `hook`, refer to it directly
   |
66 -         spa::hook::remove(*self.listener);
66 +         hook::remove(*self.listener);
   |

error[E0433]: failed to resolve: could not find `hook` in `spa`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/metadata.rs:95:14
   |
95 |         spa::hook::remove(*self.listener);
   |              ^^^^ could not find `hook` in `spa`
   |
help: consider importing one of these items
   |
4  + use crate::spa::utils::hook;
   |
4  + use spa::utils::hook;
   |
help: if you import `hook`, refer to it directly
   |
95 -         spa::hook::remove(*self.listener);
95 +         hook::remove(*self.listener);
   |

error[E0433]: failed to resolve: could not find `hook` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/node.rs:254:14
    |
254 |         spa::hook::remove(*self.listener);
    |              ^^^^ could not find `hook` in `spa`
    |
help: consider importing one of these items
    |
4   + use crate::spa::utils::hook;
    |
4   + use spa::utils::hook;
    |
help: if you import `hook`, refer to it directly
    |
254 -         spa::hook::remove(*self.listener);
254 +         hook::remove(*self.listener);
    |

error[E0433]: failed to resolve: could not find `hook` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/port.rs:193:14
    |
193 |         spa::hook::remove(*self.listener);
    |              ^^^^ could not find `hook` in `spa`
    |
help: consider importing one of these items
    |
4   + use crate::spa::utils::hook;
    |
4   + use spa::utils::hook;
    |
help: if you import `hook`, refer to it directly
    |
193 -         spa::hook::remove(*self.listener);
193 +         hook::remove(*self.listener);
    |

error[E0405]: cannot find trait `ReadableDict` in this scope
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/properties.rs:111:25
    |
111 |     pub fn from_dict<D: ReadableDict>(dict: &D) -> Self {
    |                         ^^^^^^^^^^^^ not found in this scope

error[E0405]: cannot find trait `ReadableDict` in this scope
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/properties.rs:120:6
    |
120 | impl ReadableDict for Properties {
    |      ^^^^^^^^^^^^ not found in this scope

error[E0405]: cannot find trait `WritableDict` in this scope
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/properties.rs:126:6
    |
126 | impl WritableDict for Properties {
    |      ^^^^^^^^^^^^ not found in this scope

error[E0405]: cannot find trait `ReadableDict` in this scope
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/properties.rs:212:10
    |
212 | impl<'a> ReadableDict for PropertiesRef<'a> {
    |          ^^^^^^^^^^^^ not found in this scope

error[E0433]: failed to resolve: could not find `hook` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/proxy.rs:125:14
    |
125 |         spa::hook::remove(*self.listener);
    |              ^^^^ could not find `hook` in `spa`
    |
help: consider importing one of these items
    |
4   + use crate::spa::utils::hook;
    |
4   + use spa::utils::hook;
    |
help: if you import `hook`, refer to it directly
    |
125 -         spa::hook::remove(*self.listener);
125 +         hook::remove(*self.listener);
    |

error[E0405]: cannot find trait `ReadableDict` in this scope
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:43:31
   |
43 |     pub fn bind<T: ProxyT, D: ReadableDict>(&self, object: &GlobalObject<D>) -> Result<T, Error> {
   |                               ^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `SpaResult` in crate `spa`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:67:58
   |
67 |     pub fn destroy_global(&self, global_id: u32) -> spa::SpaResult {
   |                                                          ^^^^^^^^^ not found in `spa`
   |
help: consider importing one of these items
   |
4  + use crate::spa::utils::result::SpaResult;
   |
4  + use spa::utils::result::SpaResult;
   |
help: if you import `SpaResult`, refer to it directly
   |
67 -     pub fn destroy_global(&self, global_id: u32) -> spa::SpaResult {
67 +     pub fn destroy_global(&self, global_id: u32) -> SpaResult {
   |

error[E0433]: failed to resolve: could not find `SpaResult` in `spa`
  --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:77:14
   |
77 |         spa::SpaResult::from_c(result)
   |              ^^^^^^^^^ could not find `SpaResult` in `spa`
   |
help: consider importing one of these items
   |
4  + use crate::spa::utils::result::SpaResult;
   |
4  + use spa::utils::result::SpaResult;
   |
help: if you import `SpaResult`, refer to it directly
   |
77 -         spa::SpaResult::from_c(result)
77 +         SpaResult::from_c(result)
   |

error[E0433]: failed to resolve: could not find `hook` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:112:14
    |
112 |         spa::hook::remove(*self.listener);
    |              ^^^^ could not find `hook` in `spa`
    |
help: consider importing one of these items
    |
4   + use crate::spa::utils::hook;
    |
4   + use spa::utils::hook;
    |
help: if you import `hook`, refer to it directly
    |
112 -         spa::hook::remove(*self.listener);
112 +         hook::remove(*self.listener);
    |

error[E0405]: cannot find trait `ReadableDict` in this scope
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:209:28
    |
209 | pub struct GlobalObject<D: ReadableDict> {
    |                            ^^^^^^^^^^^^ not found in this scope

error[E0405]: cannot find trait `ReadableDict` in this scope
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/registry.rs:240:9
    |
240 | impl<D: ReadableDict> GlobalObject<D> {
    |         ^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `Direction` in crate `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/stream.rs:152:25
    |
152 |         direction: spa::Direction,
    |                         ^^^^^^^^^ not found in `spa`
    |
help: consider importing one of these items
    |
6   + use crate::spa::utils::Direction;
    |
6   + use spa::utils::Direction;
    |
help: if you import `Direction`, refer to it directly
    |
152 -         direction: spa::Direction,
152 +         direction: Direction,
    |

error[E0433]: failed to resolve: could not find `hook` in `spa`
   --> /home/tonino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pipewire-0.7.2/src/stream.rs:657:14
    |
657 |         spa::hook::remove(*self.listener);
    |              ^^^^ could not find `hook` in `spa`
    |
help: consider importing one of these items
    |
6   + use crate::spa::utils::hook;
    |
6   + use spa::utils::hook;
    |
help: if you import `hook`, refer to it directly
    |
657 -         spa::hook::remove(*self.listener);
657 +         hook::remove(*self.listener);
    |

Some errors have detailed explanations: E0405, E0412, E0432, E0433.
For more information about an error, try `rustc --explain E0405`.
error: could not compile `pipewire` (lib) due to 31 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `wayland-pipewire-idle-inhibit v0.5.0`, intermediate artifacts can be found at `/tmp/cargo-installSDGnKn`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "wayland-pipewire-idle-inhibit"
-version = "0.5.0"
+version = "0.5.1"
 authors = ["Rafael Carvalho <contact@rafaelrc.com>"]
 edition = "2021"
 license = "GPL-3.0-only"
@@ -14,6 +14,8 @@ categories = ["multimedia"]
 
 [dependencies]
 pipewire = "0.7.2"
+libspa = "0.7.2"
+libspa-sys = "0.7.2"
 log = { version = "0.4", features = ["serde"] }
 env_logger = "0.10"
 timer = "0.2"

